extras/traccc_throughput_mt_profiler.sh

```diff
@@ -0,0 +1,150 @@
+   echo ""
+   echo "Detailed options:"
+   echo "  -g <detectorGeomFile> Detector geometry CSV file"
+   echo "  -d <digiConfigFile>   Detector digitization JSON file"
+   echo ""
+}
+
+# Parse the command line arguments.
+TRACCC_EXECUTABLE=${TRACCC_EXECUTABLE:-"traccc_throughput_mt"}
+TRACCC_MAX_THREADS=${TRACCC_MAX_THREADS:-$(nproc)}
+TRACCC_REPETITIONS=${TRACCC_REPETITIONS:-1}
+TRACCC_EVT_MULTI=${TRACCC_EVT_MULTI:-"1"}
+TRACCC_CSV_FILE=${TRACCC_CSV_FILE:-"output.csv"}
+TRACCC_DET_FILE=${TRACCC_DET_FILE:-"tml_detector/trackml-detector.csv"}
+TRACCC_DIGI_FILE=${TRACCC_DIGI_FILE:-"tml_detector/default-geometric-config-generic.json"}
+while getopts ":x:t:r:e:c:g:d:h" opt; do
+   case $opt in
+      x)
+         TRACCC_EXECUTABLE=$OPTARG
+         ;;
+      t)
+         TRACCC_MAX_THREADS=$OPTARG
+         ;;
+      r)
+         TRACCC_REPETITIONS=$OPTARG
+         ;;
+      e)
+         TRACCC_EVT_MULTI=$OPTARG
+         ;;
+      c)
+         TRACCC_CSV_FILE=$OPTARG
+         ;;
+      g)
+         TRACCC_DET_FILE=$OPTARG
+         ;;
+      d)
+         TRACCC_DIGI_FILE=$OPTARG
+         ;;
+      h)
+         usage
+         exit 0
+         ;;
+      :)
+         echo "Argument -$OPTARG requires a parameter!"
+         usage
+         exit 1
+         ;;
+      ?)
+         echo "Unknown argument: -$OPTARG"
+         usage
+         exit 1
+         ;;
+   esac
+done
+
+# Print the configuration received.
+echo "Using configuration:"
+echo "   EXECUTABLE  : ${TRACCC_EXECUTABLE}"
+echo "   MAX_THREADS : ${TRACCC_MAX_THREADS}"
+echo "   REPETITIONS : ${TRACCC_REPETITIONS}"
+echo "   EVT_MULTI   : ${TRACCC_EVT_MULTI}"
+echo "   CSV_FILE    : ${TRACCC_CSV_FILE}"
+echo "   DET_FILE    : ${TRACCC_DET_FILE}"
+echo "   DIGI_FILE   : ${TRACCC_DIGI_FILE}"
+
+# Check whether the output file already exists. Refuse to overwrite existing
+# files.
+if [[ -f "${TRACCC_CSV_FILE}" ]]; then
+   echo "***"
+   echo "*** Will not overwrite ${TRACCC_CSV_FILE}!"
+   echo "***"
+   exit 1
+fi
+
+# The input directories to use.
+TRACCC_INPUT_DIRS=("ttbar_mu20"  "ttbar_mu40"  "ttbar_mu60"  "ttbar_mu80"
+                   "ttbar_mu100" "ttbar_mu140" "ttbar_mu200" "ttbar_mu300")
+
+# The number of events to process for the different mu values. Chosen to take
+# roughly the same amount of time to process on a CPU.
+declare -A TRACCC_EVT_COUNT
+TRACCC_EVT_COUNT["ttbar_mu20"]=$((1000*${TRACCC_EVT_MULTI}))
+TRACCC_EVT_COUNT["ttbar_mu40"]=$((500*${TRACCC_EVT_MULTI}))
+TRACCC_EVT_COUNT["ttbar_mu60"]=$((250*${TRACCC_EVT_MULTI}))
+TRACCC_EVT_COUNT["ttbar_mu80"]=$((200*${TRACCC_EVT_MULTI}))
+TRACCC_EVT_COUNT["ttbar_mu100"]=$((150*${TRACCC_EVT_MULTI}))
+TRACCC_EVT_COUNT["ttbar_mu140"]=$((90*${TRACCC_EVT_MULTI}))
+TRACCC_EVT_COUNT["ttbar_mu200"]=$((50*${TRACCC_EVT_MULTI}))
+TRACCC_EVT_COUNT["ttbar_mu300"]=$((20*${TRACCC_EVT_MULTI}))
+
+# Put a header on the CSV file.
+echo "directory,threads,loaded_events,cold_run_events,processed_events,warm_up_time,processing_time" \
```

Comment:
This will need an update
```suggestion
echo "directory,threads,loaded_events,cold_run_events,processed_events,target_cells_per_partition,warm_up_time,processing_time" \
```

Reply:
Good catch! :+1: